### PR TITLE
switch to JSON logging by and ensure build-specific log annotations are included

### PR DESCRIPTION
**Pull request by steved**
TODOs

- [x] There are logrus messages that will need to be moved to use zap, such as: https://github.com/dominodatalab/forge/blob/d6652a5d08bf8d9b04df1319801abb7540d68e89/internal/builder/embedded/bkimage/push.go#L28
- [x] Craft a consistent format for user-readable logs from Forge
- [x] [buildkit console output](https://github.com/dominodatalab/forge/blob/master/vendor/github.com/moby/buildkit/util/progress/progressui/display.go) has a complex way of outputting user-readable logs that I need to understand in order for us to display similarly.
- [ ] ~~Tests? Specifically an E2E failure test would be nice and some unit tests on the display output.~~

### CODE REVIEW

File: internal/builder/embedded/util.go [hunk 1]
```diff
@@ -72,11 +71,15 @@ func solveRequestWithContext(sessionID string, image string, opts *config.BuildO
 	return req, nil
 }
 
-func showProgress(ch chan *controlapi.StatusResponse, noConsole bool) error {
-	displayCh := make(chan *bkclient.SolveStatus)
+func displayProgress(ch chan *controlapi.StatusResponse, progressFunc func(chan *bkclient.SolveStatus) error) error {
```
Comment (steved) [inline internal/builder/embedded/util.go:74]: I had some real trouble doing single producer, multi consumer and at some point just sort of decided that it's probably not worth it as a feature in general. If we do want it, I'll need some help with a better design.

**Comment by steved** [comment]
cc @sonnysideup

File: internal/builder/builder.go [hunk 2]
```diff
@@ -3,12 +3,15 @@ package builder
...
+	bkclient "github.com/moby/buildkit/client"
+
 	"github.com/dominodatalab/forge/internal/builder/config"
 	"github.com/dominodatalab/forge/internal/builder/embedded"
 )
 
+
 type OCIImageBuilder interface {
-	BuildAndPush(context.Context, *config.BuildOptions) ([]string, error)
+	BuildAndPush(context.Context, *config.BuildOptions, func(chan *bkclient.SolveStatus) error) ([]string, error)
```
Comment (sonnysideup) [inline internal/builder/builder.go:14]: Unfortunately, this change will binds any/every implementation to buildkit. I like the idea of using an `io.Writer` type but what do you think about this [alternative approach](https://github.com/dominodatalab/forge/pull/27) to accomplish the same thing?

File: controllers/containerimagebuild_controller.go [hunk 3]
```diff
@@ -229,3 +237,14 @@ func (r *ContainerImageBuildReconciler) updateResourceStatus(ctx context.Context
 	}
 	return nil
 }
+
+// This logs the underlying error from a build when the display channels inside builder.embedded have not yet been initialized.
+func logUnknownInstructionError(log logr.Logger, err error) {
+	if unwrappedError := errors.Unwrap(err); unwrappedError != nil {
```
Comment (sonnysideup) [inline controllers/containerimagebuild_controller.go:243]: Does `errors.Cause(errors.Cause(err))` not accomplish the same thing here w/o the nil check? Just curious.

Comment (steved) [reply controllers/containerimagebuild_controller.go:243]: So `err` in this case is a `fmt.wrapError` which wraps an error that we need the `Cause` of. It's unfortunately an issue of two different libraries. In actuality `errors.Unwrap` is just a passthrough for the stdlib errors library:
https://github.com/pkg/errors/blob/614d223910a179a466c1767a985424175c39b465/go113.go#L37

Maybe I should codify that here and call `stderrors.Unwrap` as well?

**Pull request merged**